addon/routes/list-form.js

```diff
@@ -61,29 +61,26 @@ export default ProjectedModelFormRoute.extend(PaginatedRouteMixin, SortableRoute
     Ember.assert('page must be greater than zero.', page > 0);
     Ember.assert('perPage must be greater than zero.', perPage > 0);
 
-    let store = this.store;
     let modelName = this.get('modelName');
-    let adapter = store.adapterFor(modelName);
-
-    let pageQuery = adapter.getPaginationQuery(page, perPage);
-
+    let projectionName = this.get('modelProjection');
+    let serializer = this.store.serializerFor(modelName);
     let sorting = this.deserializeSortingParam(params.sort);
-    let sortQuery = adapter.getSortingQuery(sorting, store.serializerFor(modelName));
-
-    let modelClass = this.store.modelFor(this.get('modelName'));
-    let proj = modelClass.projections.get(this.get('modelProjection'));
-    let filterString = this.getFilterString(proj, params);
-    let limitFunctionQuery = adapter.getLimitFunctionQuery(filterString);
 
-    let query = {};
-    Ember.merge(query, pageQuery);
-    Ember.merge(query, sortQuery);
-    Ember.merge(query, limitFunctionQuery);
-    Ember.merge(query, { projection: this.get('modelProjection') });
+    let builder = new QueryBuilder(this.store)
+      .from(modelName)
+      .selectByProjection(projectionName)
+      .top(perPage)
+      .skip((page - 1) * perPage)
+      .count()
+      .orderBy(
+        sorting
+          .map(i => `${serializer.keyForAttribute(i.propName)} ${i.direction}`)
+          .join(',')
+      );
```

Comment:
No additional filtering is applied in current implementation. This will break existing calls to 'list-form' route with 'lf' and 'filter' URL-params.